web/packages/teleport/src/Assist/Chat/Chat.tsx

```diff
@@ -89,9 +100,28 @@ export function Chat() {
 
   const handleSubmit = useCallback(
     (message: string) => {
-      send(message);
+      send(message).then(() => {
+        if (messages.length == 1) {
+          // Use the second message/first message from a user to generate the title.
+          (async () => {
+            // Generate title using the last message and OpenAI API.
+            const title = await generateTitle(message);
+            // Set the title in the backend.
+            await setConversationTitle(props.conversationId, title);
+            // Update the title in the frontend.
+            setConversations(conversations =>
+              conversations.map(c => {
```

Comment:
Why are we using `map` here? Can we replace this with `find` or `findIndex`.

Reply:
`map` here is correct - it sets the conversations to a new array, replacing the title in the relevant conversation object

Reply:
ah, gotcha. I forget about all of the react funkiness in how it does state.